File: date/d250108/python/person.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def greet(self, greeting="Hej"):
        return f"{greeting}, jag heter {self.name} och är {self.age} år gammal."


p = Person("Anna", 30)
print(p.greet())
print(p.greet("Hallå"))

File: DEVOPS24/Person.java
package d250108;

class Person {
    String name;
    int age;

    Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    String greet() {
        // Alt. sätt men med duplicerad kod: return "Hej" + ", jag heter " + name + " och är " + age + " år gammal.";
        return greet("Hej");
    }

    String greet(String greeting) {
        return greeting + ", jag heter " + name + " och är " + age + " år gammal.";
    }

    public static void main(String[] args) {
        Person p = new Person("Anna", 30);
        System.out.println(p.greet());
        System.out.println(p.greet("Hallå"));
    }
}
